### PR TITLE
Clockcult Removal 2019

### PR DESCRIPTION
**Pull request by KomradeSpectre**
## About The Pull Request

**Removes Clock Cult entirely from the code, excluding the assets.**

**Maintainer permission for the removal during the freeze below, the tweak themselves count as crew content tweaks if anything.**
image

Moves the spinning chair functionality from the deleted Brass chair over to the Bronze variant to preserve its aesthetic functionality, though I can change this if it is a point of contention.

Finally, fixes a Lavaland ruin to use the bronze metal variants of structs and turfs, and uses the null rod variant "Ancient Spear" to replace the missing Clock Cult spear the ruin uses (they're pretty much 1:1, but I can remove this too if it becomes a point of contention). The previous clockcult component items that were in this ruin before are now a fluff structure as requested by @ExcessiveUseOfCobblestone.

Based off most of the work done by ShizCalev here
[#38163 **Deletes Clock Cult by ShizCalev**](https://github.com/tgstation/tgstation/pull/38163)

## Why It's Good For The Game

The Clock Cult mode has sat unmaintained by its developer **(who had left the community)**, unsupported **(dropped by maintainers)**, out of rotation, and extremely intrusive to most aspects of the game **(look at the files changed to see how many systems it was in)**. On top of this it has major design flaws regarding either balance, playability, or otherwise that have existed for both its lifespan and after.

In the long run, it would be far better for someone to take the assets (which still exist) and remake the mode from a blank slate if they were to, rather than build upon an extremely shaky foundation like what I had seen when I was manually applying Shiz' removals.

## Changelog
:cl: ShizCalev, TheMythicGhost
del: Nanotrasen's Department of Supernatural Defense fires the newly built Wave Motion Gun at Reebe, destroying the Clock Cult.
tweak: After reverse engineering technologies found after the annihilation of Reebe, Bronze chairs now have incorporated Ratvarian A E S T H E T I C auto-spinning technology.
tweak: Due to the annihilation of Reebe, the clockwork structures and turfs that fell to lavaland alongside Ratvar's corpse have lost their power becoming mere bronze, but the spear seems to have been touched by holy energies.
/:cl:

### CODE REVIEW

**Comment by KomradeSpectre** [comment]
Oopsie poopsies, got to fix a simple mess up.

**Comment by ATH1909** [comment]
DinkleNH is gonna be saddened by this; he really liked clock cult (and I do too, when it's played in moderation).

**Comment by Shadowflame909** [comment]
tis not the only dead mode in the game

Theres devil, monkey, meteors, speedy revs, never-ending revs, clown cult.

Why remove this one specifically KomradeSpectre?

Especially now that reebee doesn't spawn unless the mode is enabled. So it isn't causing any lag either.

**Comment by DaedalusGame** [comment]
@Shadowflame909 #45316

**Comment by Iatots** [comment]
You'd think monkey mode would be the first to go. Is it that deeply rooted in the code?

**Comment by Whoneedspacee** [comment]
clock cult is fucking everywhere and its just not a good mode and people have had plenty of time to make any attempt to fix it

**Comment by GuyonBroadway** [comment]
It's a shame, it'd be nice to keep the assets for adminbus territory but having to reload a whole fucking Z-level just for admeems is pretty shitty.


Clock cult deserved much better than it ever got.

**Comment by PKPenguin321** [comment]
i could have sworn we already did this

**Comment by KomradeSpectre** [comment]
> 
> 
> i could have sworn we already did this

We did try, but Shiz gave up either due to not having enough time to deal with Travis, or Travis just being generally fucky back around when he tried to remove Clock Cult, he did mention that his branch was still there if anyone wanted to finish the job, so I decided to finish the job with maintainers blessings, if that answers any questions.

**Comment by Repukan** [comment]
Based

**Comment by MCterra10** [comment]
R.I.P. AuStation still runs clock cult occasionally and it is enjoyed by most of our players. You will be missed

**Comment by Shadowflame909** [comment]
I think the mode is as balanced as the current blood cult and its even more balanced then changeling

but that's just my take on it

**Comment by Taylorkills2** [comment]
Very sad to see this being removed, though I agree that for Clock Cult to be a fully functional and balanced game-mode it'd need heavy revisions it simply won't get here.
Although, funnily enough, clock-cult is actually pretty balanced(if moderately underpowered) without Reebee.
I highly doubt anyone will stop this, so I'm glad I used an antag token for it while it was still here.

**Comment by necromanceranne** [comment]
Dinkle didn't die for this!

**Comment by Grazyn** [comment]
It was long overdue.

**Comment by kriskog** [comment]
Clock cult has found its way into 192 files, not counting sprites or sounds. Removal is good for the maintainability of the rest of the game.

**Comment by Tlaltecuhtli** [comment]
can you make skewers a not clock cult trap and add it as craftable? everything else isnt important

**Comment by bobbahbrown** [comment]
down thumbers are delaying the inevitable

**Comment by KomradeSpectre** [comment]
> 
> 
> can you make skewers a not clock cult trap and add it as craftable? everything else isnt important

wyci

**Comment by 81Denton** [comment]
cockcult

Oof ok I'm done for now

**Review by 81Denton — DISMISSED** [review]
I couldn't find any glaring issues, so I'm fine with this PR. I'd appreciate it if another maintainer gave it a look as well since it touches a whopping 192 files.

File: code/game/objects/items/melee/misc.dm [hunk 1]
```diff
@@ -542,7 +542,7 @@
 /obj/item/melee/roastingstick/Initialize()
 	. = ..()
 	if (!ovens)
-		ovens = typecacheof(list(/obj/singularity, /obj/machinery/power/supermatter_crystal, /obj/structure/bonfire, /obj/structure/destructible/clockwork/massive/ratvar))
```
Comment (nemvar) [inline code/game/objects/items/melee/misc.dm]: single biggest lost.

Comment (KomradeSpectre) [reply code/game/objects/items/melee/misc.dm]: I may have made a mistake there.

Comment (KomradeSpectre) [reply code/game/objects/items/melee/misc.dm]: Actually no we good, it's good.

Comment (KomradeSpectre) [reply code/game/objects/items/melee/misc.dm]: Though if someone wants to change that to `/obj/structure/dead_ratvar` by all means.

**Comment by boodaliboo** [comment]
i dont even play tg but from what ive heard most tg players enjoy when they get a clock cult event

**Comment by CRITAWAKETS** [comment]
A small question i have is why go through the pain of removing it by basically checking every file when in that time you probably could just fix a lot of the issues the mode has?

**Comment by KomradeSpectre** [comment]
> 
> 
> A small question i have is why go through the pain of removing it by basically checking every file when in that time you probably could just fix a lot of the issues the mode has?
> 
> Doesn't help that clock cult is used in many admeme scenarios and also is generally a fun mode, even if unbalanced.

As the person that manually checked the files and implemented Shiz' changes, I can tell you without any doubt that the only way to fix the core of the issue was to remove it. From here, it can literally be build on a more steady foundation where it isn't in a million files for no fucking reason other than for special Ratvar Singularity reactions, or servant checks.

**Comment by SkeletalElite** [comment]
Did you remove the dead ratvar from the lavaland ruin too? Or just make all the stuff bronze instead of brass. Because even though the mode is gone I'd like to see the dead ratvar remain as an object thats there.

**Comment by KomradeSpectre** [comment]
> 
> 
> Did you remove the dead ratvar from the lavaland ruin too? Or just make all the stuff bronze instead of brass. Because even though the mode is gone I'd like to see the dead ratvar remain as an object thats there.

There is a map diff bot.

**Comment by KomradeSpectre** [comment]
Fixed redundant cells in the lavaland ruin by populating them with either mineral chunks, or assorted destruction clutter that makes sense. Otherwise you'd see 5 or so cells with the same contents and that's no good.

**Comment by ExcessiveUseOfCobblestone** [comment]
did you convert the old stuff into structures a la hand of god?

**Comment by dee-dubya** [comment]
> 
> 
> down thumbers are delaying the inevitable, the improved maintainability from removing clockcult is good
> 
> alternative for those who dont want to remove it is to readd it after its removed as a much modular implementation

That's the thing. If clock cult already hasn't been touched in close to a year now as it currently stands, the community knows nobody will bother to re-add it in a more modular format. So it'll be gone for good, and people don't want that. They want clock cult to stay, but nobody is willing and able to work out the kinks to put it back in rotation.

**Comment by Repukan** [comment]
Someone should make a PR to add straws so they can suck it up

**Comment by kriskog** [comment]
> 
> 
> > down thumbers are delaying the inevitable, the improved maintainability from removing clockcult is good
> > alternative for those who dont want to remove it is to readd it after its removed as a much modular implementation
> 
> That's the thing. If clock cult already hasn't been touched in close to a year now as it currently stands, the community knows nobody will bother to re-add it in a more modular format. So it'll be gone for good, and people don't want that. They want clock cult to stay, but nobody is willing and able to work out the kinks to put it back in rotation.

Bold assumption. Maybe this is the kick they need to do so.

**Comment by KomradeSpectre** [comment]
Everything should be well in order, demands have been met regarding the ruin.

**Comment by SpaceManiac** [comment]
> devil, monkey, meteors, speedy revs, never-ending revs, clown cult.

for the record both devil and meteors are viable on small downstreams and none of the other modes listed have anywhere near the code volume and spread as clockcult

**Comment by Shadowflame909** [comment]
So is it about making it compile faster?

**Comment by 81Denton** [comment]
Headmins have agreed on removing Clock Cult, so I'm going to go ahead and merge this.

I'm certain that some bugs might pop up since this change affects a whopping 193 files, but it'll be nothing we can't fix.

**Comment by optimumtact** [comment]
just to clarify, the headadmins were asked if they were okay with the idea of us removing it, not if they wanted it removed

**Comment by KomradeSpectre** [comment]
**A song to go with this removal that I should have posted earlier, as it's good music.**
[](https://www.youtube.com/watch?v=M3mdHmhI3cs)

**Comment by Magikarpet** [comment]
If you're upset and you want clockcult back start from the ground up

the code is unmaintainable and everywhere🤢

**Comment by MCterra10** [comment]
F you will be missed, but it was for the best

**Comment by wesoda25** [comment]
smile

**Pull request merged**